pkg/conversion/mapi2capi/aws_test.go

```diff
@@ -16,43 +16,140 @@ limitations under the License.
+		ARN: ptr.To("arn:aws:ec2:us-east-1::image/ami-1234567890abcdef0"),
+	}))
+	awsMAPIMachineWithAMIFilter = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithAMI(mapiv1.AWSResourceReference{
+		Filters: []mapiv1.Filter{{Name: "name", Values: []string{"test"}}},
+	}))
+	awsMAPIMachineWithMissingAMI      = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithAMI(mapiv1.AWSResourceReference{}))
+	awsMAPIMachineWithInvalidMetadata = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithMetadataServiceOptions(mapiv1.MetadataServiceOptions{
+		Authentication: "unsupported",
+	}))
+	awsMAPIMachineWithMissingVolumeSize = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{EBS: &mapiv1.EBSBlockDeviceSpec{}},
+	}))
+	awsMAPIMachineWithNonRootVolumeNotDeleted = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{DeviceName: ptr.To("/dev/sdb"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10)), DeleteOnTermination: ptr.To(false)}},
+	}))
+	awsMAPIMachineWithNoDevice = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{NoDevice: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10))}},
+	}))
+	awsMAPIMachineWithVirtualName = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10))}},
+	}))
+	awsMAPIMachineWithVirtualNameAndRootVolNotDeletedOnTerm = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10)), DeleteOnTermination: ptr.To(false)}},
+	}))
+	awsMAPIMachineWithVirtualNameAndMissingEBSConfig = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test")},
+	}))
+	awsMAPIMachineWithMissingEBSConfig = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{DeviceName: ptr.To("/dev/sdb")},
+	}))
+	awsMAPIMachineWithRootVolumeNotDeleted = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10)), DeleteOnTermination: ptr.To(false)}},
+	}))
+	awsMAPIMachineWithNoDeviceAndVirtualName = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test"), NoDevice: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10))}},
+	}))
```

Comment:
Can you add some whitespace in here, this becomes quite a wall as it stands

Out of interest, are they all based on a single base machine?

Would it be easier to create providerSpecs and pass a PS builder/built PS into the test case and have that substituted into the machine within the test case?

Reply:
Yes let's do this in the `Entry` rather than here, it's easier to have these two details close together.